test/gpu/native/studies/shoc/result.chpl

```diff
@@ -22,112 +22,125 @@ module ResultDB {
         var atts: string;
         var attsSuffix: string;
         var results: sortedMap;
+        var homeLocale: locale;
 
         proc init(testName: string, units: string, atts = "BlckSz:", attsSuffix = "KB") {
             this.testName = testName;
             this.units = units;
             this.atts = atts;
             this.attsSuffix = attsSuffix;
             this.results = new sortedMap(string, list(real));
+            this.homeLocale = here;
         }
 
         proc addToDatabase(blockSize: string, value: real){
-            // Since the [] Access adds the mapping if it does not
-            // exist we do not have to initialize a new list when one
-            // does not exist which is neat
-            this.results[blockSize].append(value);
+            on homeLocale {
```

Comment:
Oh, you have to switch this back to the CPU locale. I was running into this with my [`/test/gpu/native/array_on_device/studies/shoc/triadAOD.chpl`](https://github.com/chapel-lang/chapel/blob/main/test/gpu/native/array_on_device/studies/shoc/triadAOD.chpl) test.

I ended up commenting out the calls to `addToDatabase` out but I wonder if they would work now with this change: https://github.com/chapel-lang/chapel/blob/0fd6a99016b6de5e8f548590661c600a0a7b1035/test/gpu/native/array_on_device/studies/shoc/triadAOD.chpl#L172

Anyway, you can leave that to me if you want but I'm glad I noticed that you changed it.

---


I also kind of wish we could could do something like: `here.homeLocale` or `here.cpuLocale` instead of having to cache it when we initialize the module.

If you have the time/inclination I'd create a Github issue for that as a potential feature request and cite this example here as something to update if we do decide to modify it.

Reply:
Yeah, I found the need for that feature in other contexts, too (can't remember where exactly). I'll fire off an issue for this.

Reply:
https://github.com/chapel-lang/chapel/issues/21754